Formula/haskell-language-server.rb

```diff
@@ -0,0 +1,29 @@
+class HaskellLanguageServer < Formula
+  desc "Integration point for ghcide and haskell-ide-engine. One IDE to rule them all"
+  homepage "https://github.com/haskell/haskell-language-server"
+  url "https://github.com/haskell/haskell-language-server/archive/1.0.0.tar.gz"
+  sha256 "14e28d6621d029f027fae44bc4a4ef62c869dab24ff01b88a2e51e6679cbff6c"
+  license "Apache-2.0"
+  head "https://github.com/haskell/haskell-language-server.git"
+
+  # we need :github_latest here because otherwise livecheck picks up spurious non-release tags
+  livecheck do
+    url :stable
+    strategy :github_latest
+  end
+
+  depends_on "cabal-install" => :build
+  depends_on "ghc" => :build
```

Comment:
```suggestion
  depends_on "cabal-install" => [:build, :test]
  depends_on "ghc" => [:build, :test]
```
Looks like you need these available for the test to work. Squash this change again with the first one. Thanks!

Reply:
Thanks, didn't know that was needed 😅

Reply:
Just to make sure, will the language server work with any version of ghc installed, even if it was built with a different version?

Reply:
As also discussed here: https://github.com/Homebrew/discussions/discussions/386, the `haskell-language-server` will only work with projects using exactly the same version of `ghc` as one used to compile it.
That is why HLS maintainers release pre-built binaries compiled against several GHC versions: https://github.com/haskell/haskell-language-server/releases/tag/1.0.0 (scroll to Assets).

That is why I initially proposed a cask with full release: https://github.com/Homebrew/homebrew-cask/pull/100527. When it was rejected I submitted this/current PR. But the downside is that HLS _formula_ will only work with GHC 8.10.4 (current `ghc` formula version - once `ghc` formula is updated to new version, `hls` formula will need to be updated and rebuilt. But I thought that since `ghc` is also only "a version" then `hls` might be a compatible one in Homebrew space.

Then, my other line of thought is to propose HLS team to maintain their own tap with the prebuilt binaries, perhaps split into wrapper and each-GHC-compiled-version separately... since the binaries are already built by the HLS maintainers and users could pick and choose whichever version they require.

Reply:
Ah, I see, I missed that part of the discussion. If this is the case then we should probably make `ghc` an unconditional dependency (i.e. not just build-only). Unless you think it's valuable for users to be able to install this and use it with their own ghc 8.10.4?

Reply:
HLS needs one of `ghc`, `cabal-install` or `haskell-stack`, whichever one uses for their project... Since the Homebrew Way and the Haskell Way seem to be a bit unaligned I would treat Homebrew formulas as good start for the beginners - install recent GHC and matching HLS version and get hacking.
...I am under impression that "real Haskellers" just compile HLS themselves when they need another version for a project (from https://github.com/haskell/haskell-language-server/issues/703#issuecomment-765672788, and from blogs, reddit - since I am a Haskell beginner myself and don't yet comprehend its build ecosystem fully).

Reply:
Hmmm, it might be useful to include a `caveats` method then. Something like
```ruby
def caveats
  versioned_ghc = "ghc #{Formula["ghc"].version}"

  <<~EOS
    #{name} is built for #{versioned_ghc}. You will need to provide
    your own #{versioned_ghc} or install one with
      brew install ghc
  EOS
end
```
Feel free to tweak the wording a bit.

Reply:
Cool, thanks. I think the wording is good. Just added a link to HLS prerequisites docs.